earthaccess/search.py

```diff
@@ -468,8 +468,8 @@ def get(self, limit: int = 2000) -> List[DataGranule]:
             RuntimeError: The CMR query failed.
         """
         response = get_results(self.session, self, limit)
-
-        cloud = self._is_cloud_hosted(response[0])
+        if response:
+            cloud = self._is_cloud_hosted(response[0])
```

Comment:
Although the expression in the return statement will still work even when `cloud` is undefined (when there are no results), perhaps we should make sure it is always defined.

```suggestion
        cloud = response and self._is_cloud_hosted(response[0])
```